submissions/kt-std/object-oriented-js/app.js

```diff
@@ -0,0 +1,255 @@
+ * If Player still left lives and it has crossed the border of any Enemy,
+ * then decrease Player lives amount, delete live item from the LIVES_CONTAINER
+ * and reset Player position to the initial state.
+ */
+Player.prototype.update = function () {
+    if (!this.lives) {
+        displayMessage("lose", "flex", messages);
+    } else {
+        allEnemies.forEach((enemy) => {
+            if (
+                Math.abs(this.x - enemy.x) < X_GAP &&
+                Math.abs(this.y - enemy.y) < Y_GAP
+            ) {
+                this.lives--;
+                removeLivesFromContainer();
+                resetPosition(this, INITIAL_X, INITIAL_Y);
+            }
+        });
+    }
+};
+
+/* This function resets Player lives to the initial state
+ * and calls the function to append lives to the LIVES_CONTAINER
+ */
+Player.prototype.resetLives = function (livesAmount) {
+    this.lives = livesAmount;
+    appendLivesToContainer(this.lives);
+};
+
+/* This function handles user input and moves Player to the
+ * special position depending on condition. If user reached the first(finish) row
+ * it displays greeting message. If Player reached the left/right canvas edge it
+ * transfers Player to the oposite canvas side. If Player reached canvas bottom it
+ * remains on its place.
+ */
+Player.prototype.handleInput = function (key) {
+    this.move(key);
+    if (!this.y) {
+        displayMessage("win", "flex", messages);
+    } else if (this.y > INITIAL_Y) {
+        this.y = INITIAL_Y;
+    } else if (this.x >= CANVAS_WIDTH) {
+        this.x = CANVAS_START;
+    } else if (this.x < CANVAS_START) {
+        this.x = CANVAS_WIDTH - X_STEP;
+    } else if (this.y < CANVAS_START) {
+        this.y = CANVAS_START;
+    }
+};
+
+/*
+ * This function sets Player coordinate depending on the key pressed
+ */
+Player.prototype.move = function (key) {
+    switch (key) {
+        case "up":
+            this.y -= Y_STEP;
+            break;
+        case "down":
+            this.y += Y_STEP;
+            break;
+        case "left":
+            this.x -= X_STEP;
+            break;
+        case "right":
+            this.x += X_STEP;
+            break;
+    }
+};
+
+document.addEventListener("keyup", function (e) {
+    const allowedKeys = {
+        37: "left",
+        38: "up",
+        39: "right",
+        40: "down",
+    };
+    player.handleInput(allowedKeys[e.keyCode]);
+});
+
+/*
+ * This function resets Player and Enemies position, Player's lives
+ * and Enemies speed to restart the game if the reset button is pressed
+ */
+document.querySelector("body").addEventListener("click", function (e) {
+    if (e.target.id === "reset") {
+        displayMessage(e.target.offsetParent.id, "none");
+        resetPosition(player, INITIAL_X, INITIAL_Y);
+        player.resetLives(LIVES_AMOUNT);
+        allEnemies.map((enemy, index) => {
+            resetPosition(enemy, CANVAS_START, Y_STEP * (index + 1));
```

Comment:
if we move this function to parent class Actor then we can use just like this:

``
enemy.setPosition(CANVAS_START, Y_STEP * (index + 1));
``